EligibilityCheckOrchestrator/index.ts

```diff
@@ -187,18 +188,44 @@ export const handler = function*(
   const maybeMessageType = getMessageType(validatedEligibilityCheck);
 
   if (isSome(maybeMessageType)) {
-    yield context.df.callActivityWithRetry(
-      "SendMessageActivity",
-      retryOptions,
-      SendMessageActivityInput.encode({
-        checkProfile: false,
-        content: getMessage(
-          maybeMessageType.value,
-          eligibilityCheckResponse.validBefore
-        ),
-        fiscalCode: eligibilityCheckResponse.fiscalCode
-      })
-    );
+    if (maybeMessageType.value === "EligibilityCheckConflict") {
+      const isBonusProcessingRunning = yield context.df.callActivityWithRetry(
+        "CheckBonusStatusRunningActivity",
+        retryOptions,
+        CheckBonusStatusActivityInput.encode({ fiscalCode: orchestratorInput })
+      );
+      const content = isBonusProcessingRunning
+        ? getMessage(
+            maybeMessageType.value,
+            eligibilityCheckResponse.validBefore
+          )
+        : getMessage(
+            "EligibilityCheckConflictWithBonusActivated",
+            eligibilityCheckResponse.validBefore
+          );
+      yield context.df.callActivityWithRetry(
+        "SendMessageActivity",
+        retryOptions,
+        SendMessageActivityInput.encode({
+          checkProfile: false,
+          content,
+          fiscalCode: eligibilityCheckResponse.fiscalCode
+        })
+      );
+    } else {
+      yield context.df.callActivityWithRetry(
+        "SendMessageActivity",
+        retryOptions,
+        SendMessageActivityInput.encode({
+          checkProfile: false,
+          content: getMessage(
+            maybeMessageType.value,
+            eligibilityCheckResponse.validBefore
+          ),
+          fiscalCode: eligibilityCheckResponse.fiscalCode
+        })
+      );
+    }
```

Comment:
We need some test to cover all introduced branches

Reply:
done